### PR TITLE
Update seastar submodule

### PR DESCRIPTION
**Pull request by m-szymon**
Includes "http: allow control over default response headers" (scylladb/seastar#3217) which adds set_server_header() and set_generate_date_header() APIs needed by PR #28288.

* seastar 4d268e0e...6c4dcd18 (36):
  > apps/httpd: replace deprecated reply::done() with write_body()
  > missing header(s)
  > net: Fix missing throw for runtime_error in create_native_net_device
  > tests/io_queue: account for token bucket refill granularity in bandwidth checks
  > Merge 'iovec: fix iovec_trim_front infinite loop on zero-length iovecs' from Travis Downs
    tests: add regression tests for zero-length iovec handling
    iovec: fix iovec_trim_front infinite loop on zero-length iovecs
  > util/process: graduate process management API from experimental
  > cooking: don't register ready.txt as a build output
  > sstring: make make_sstring not static
  > Add SparkyLinux to debian list in install-dependencies.sh
  > http: allow control over default response headers
  > Merge 'chunked_fifo: make cached chunk retention configurable' from Brandon Allard
    tests/perf: add chunked_fifo microbenchmarks
    chunked_fifo: set the default free chunk retention to 0
    chunked_fifo: make free chunk retention configurable
  > Merge 'reactor_backend: fix pollable_fd_state_completion reuse in io_uring' from Kefu Chai
    tests: add regression test for pollable_fd_state_completion reuse
    reactor_backend: use reset() in AIO and epoll poll paths
    reactor_backend: fix pollable_fd_state_completion reuse after co_await in io_uring
  > Merge 'coroutine: Generator cleanups' from Kefu Chai
    coroutine/generator: extract schedule_or_resume helper
    coroutine/generator: remove unused next_awaiter classes
    coroutine/generator: remove write-only _started field
    coroutine/generator: assert on unreachable path in buffered await_resume
    coroutine/generator: add elements_of tag and #include <ranges>
    coroutine/generator: add empty() to bounded_container concept
  > cmake: bump minimum Boost version to 1.79.0
  > seastar_test: remove unnecessary headers
  > cmake: bump minimum GnuTLS version to 3.7.4
  > Merge 'reactor: add get_all_io_queues() method' from Travis Downs
    tests: add unit test for reactor::get_all_io_queues()
    reactor: add get_all_io_queues() method
    reactor: move get_io_queue and try_get_io_queue to .cc file
  > http: deprecate reply::done(), remove _response_line dead field
  > core: Deprecate scattered_message
  > ci: add workflow dispatch to tests workflow
  > perf_tests: exit non-zero when -t pattern matches no tests
  > Replace duplicate SEGV_MAPERR check in sigsegv_action() with SEGV_ACCERR.
  > perf_tests: add total runtime to json output
  > Merge 'Relax large allocation error originating from json_list_template' from Robert Bindar
    implement move assignment operator for json_list_template
    json_list_template copy assignment operator reserves capacity upfront
  > perf_tests: add --no-perf-counters option
  > Merge 'Fix to_human_readable_value() ability to work with large values' from Pavel Emelyanov
    memory: Add compile-time test for value-to-human-readable conversion
    memory: Extend list of suffixes to have peta-s
    memory: Fix off-by-one in suffix calculation
    memory: Mark to_human_readable_value() and others constexpr
  > http: Improve writing of response_line() into the output
  > Merge 'websocket: add template parameter for text/binary frame mode and implement client-side WebSocket' from wangyuwei
    websocket: add template parameter for text/binary frame mode
    websocket: impl client side websocket function
  > file: Fix checks for file being read-only
  > reactor: Make do_dump_task_queue a task_queue method
  > Merge 'Implement fully mixed mode for output_stream-s' from Pavel Emelyanov
    tests/output_stream: sample type patterns in sanitizer builds
    tests/output_stream: extend invariant test to cover mixed write modes
    iostream: allow unrestricted mixing of buffered and zero-copy writes
    tests/output_stream: remove obsolete ad-hoc splitting tests
    tests/output_stream: add invariant-based splitting tests
    iostream: rename output_stream::_size to ::_buffer_size
  > reactor_backend: replace virtual bool methods with const bool_class members
  > resource: Avoid copying CPU vector to break it into groups
  > perf_tests: increase overhead column precision to 3 decimal places
  > Merge 'Move reactor::fdatasync() into posix_file_impl' from Pavel Emelyanov
    reactor: Deprecate fdatasync() method
    file: Do fdatasync() right in the posix_file_impl::flush()
    file: Propagate aio_fdatasync to posix_file_impl
    reactor: Move reactor::fdatasync() code to file.cc
    reactor,file: Make full use of file_open_options::durable bit
    file: Add file_open_options::durable boolean
    file: Account io_stats::fsyncs in posix_file_impl::flush()
    reactor: Move _fsyncs counter onto io_stats
  > http: Remove connection::write_body()

No backport - needed for new features.

### CODE REVIEW

**Comment by m-szymon** [comment]
Shouldn't it trigger build in CI?

**Comment by nyh** [comment]
@scylladb/releng why doesn't the CI build Scylla and run tests? It should...

**Comment by m-szymon** [comment]
@scylladbbot trigger-ci

**Comment by nyh** [comment]
Added this Seastar update.

@avikivity, I have a procedural comment/question:

I seem to remember that in the past you asked that we do **not** merge Seastar updates from pull requests (like this one) with scripts/pull_github_pr.sh. Can you please remind me why? I think it would have done the right thing - update the submodule pointer to the exact one tested in this PR. Why is this bad?

Anyway, the alternative is that I as a maintainer use `scripts/refresh-submodules.sh seastar` myself and push the result immediately. However, this has a problem: Seastar has some newer patches that weren't tested by this PR, but the refresh-submodules.sh script will take the newest patches automatically.  I solved this problem by patching scripts/refresh-submodule.sh to replace
```
bump_to=origin/${branch}
```
by
```
bump_to=${BUMP-origin/${branch}}
```

i.e.
```patch
diff --git a/scripts/refresh-submodules.sh b/scripts/refresh-submodules.sh
index abc13803ed..1d5842cfe3 100755
--- a/scripts/refresh-submodules.sh
+++ b/scripts/refresh-submodules.sh
@@ -25,7 +25,7 @@ submodules=(
 for ent in "${@:-${submodules[@]}}"; do
     submodule=${ent%%:*}
     [ ${submodule} == ${ent} ] && branch="master" || branch=${ent#*:}
-    bump_to=origin/${branch}
+    bump_to=${BUMP-origin/${branch}}
 
     pushd .
     cd $submodule
```

And then I used
```
BUMP=6c4dcd18 scripts/refresh-submodules.sh seastar
```
Where 6c4dcd18 is the exact commit that the pull request included.
This allowed me to update Seastar to exactly the commit tested by this PR.

How do you usually update the Seastar submodule (you do this much more often then I do)? Do you think this "BUMP" trick is useful and I should send a patch with it?
CC @scylladb/scylla-maint

**Comment by nyh** [comment]
Merged to next: 2943d30b0c747dbf0354c276bb9baaf17a679677
I forgot to add a "Closes" so please close this PR manually when this change reaches master.

**Comment by avikivity** [comment]
> Added this Seastar update.
> 
> @avikivity, I have a procedural comment/question:
> 
> I seem to remember that in the past you asked that we do **not** merge Seastar updates from pull requests (like this one) with scripts/pull_github_pr.sh.


I did not.

> Can you please remind me why? I think it would have done the right thing - update the submodule pointer to the exact one tested in this PR. Why is this bad?

It's not bad. I often do.

> 
> Anyway, the alternative is that I as a maintainer use `scripts/refresh-submodules.sh seastar` myself and push the result immediately. However, this has a problem: Seastar has some newer patches that weren't tested by this PR, but the refresh-submodules.sh script will take the newest patches automatically. I solved this problem by patching scripts/refresh-submodule.sh to replace
> 
> ```
> bump_to=origin/${branch}
> ```
> 
> by
> 
> ```
> bump_to=${BUMP-origin/${branch}}
> ```
> 
> i.e.
> 
> ```diff
> diff --git a/scripts/refresh-submodules.sh b/scripts/refresh-submodules.sh
> index abc13803ed..1d5842cfe3 100755
> --- a/scripts/refresh-submodules.sh
> +++ b/scripts/refresh-submodules.sh
> @@ -25,7 +25,7 @@ submodules=(
>  for ent in "${@:-${submodules[@]}}"; do
>      submodule=${ent%%:*}
>      [ ${submodule} == ${ent} ] && branch="master" || branch=${ent#*:}
> -    bump_to=origin/${branch}
> +    bump_to=${BUMP-origin/${branch}}
>  
>      pushd .
>      cd $submodule
> ```
> 
> And then I used
> 
> ```
> BUMP=6c4dcd18 scripts/refresh-submodules.sh seastar
> ```
> 
> Where 6c4dcd18 is the exact commit that the pull request included. This allowed me to update Seastar to exactly the commit tested by this PR.
> 
> How do you usually update the Seastar submodule (you do this much more often then I do)? Do you think this "BUMP" trick is useful and I should send a patch with it? CC @scylladb/scylla-maint

**Comment by nyh** [comment]
> > Added this Seastar update.
> > @avikivity, I have a procedural comment/question:
> > I seem to remember that in the past you asked that we do **not** merge Seastar updates from pull requests (like this one) with scripts/pull_github_pr.sh.
> 
> I did not.

Sorry. I don't know why I remembered that. Next time I'll just merge Seastar-update pull requests just like normal pull requests instead of jumping through hoops doing it manually. Thanks.

**Comment by avikivity** [comment]
Revert queued: b9ce78f791e4a1c48ff74979dc9364cdc336ebd1

**Comment by nyh** [comment]
> Revert queued: [b9ce78f](https://github.com/scylladb/scylladb/commit/b9ce78f791e4a1c48ff74979dc9364cdc336ebd1)

You can't just revert this Seastar update, there's also a Scylla patch merged that depends on this :-(

Since we're already in a merge freeze that you declared, wouldn't it be more efficient to just fix the bug, if you know what it is?

**Comment by avikivity** [comment]
> > Revert queued: [b9ce78f](https://github.com/scylladb/scylladb/commit/b9ce78f791e4a1c48ff74979dc9364cdc336ebd1)
> 
> You can't just revert this Seastar update, there's also a Scylla patch merged that depends on this :-(
> 

Reverted that too.

> Since we're already in a merge freeze that you declared, wouldn't it be more efficient to just fix the bug, if you know what it is?

No. It's most efficient to go back to known-good state than to enter a loop of fix-compile-test on the path that blocks everyone from making progress.
